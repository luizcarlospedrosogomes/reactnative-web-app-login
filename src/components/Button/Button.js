import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableHighlight,
} from 'react-native';
import { shadow } from '../../helpers';

class Button extends Component {
  render() {
    const { onClick, children } = this.props;

    return (
      <TouchableHighlight
        onPress={onClick}
        style={styles.button}
        underlayColor={'#0A84D0'}
      >
        <Text style={styles.buttonText}>{children}</Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  button: {
    borderRadius: 3,
    padding: 20,
    marginVertical: 10,
    marginTop: 10,
    backgroundColor: '#1B95E0',
    ...shadow,
  },
  buttonText: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 16,
  },
});

export default Button;
