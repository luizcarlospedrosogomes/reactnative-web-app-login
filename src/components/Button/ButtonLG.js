import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableHighlight,
} from 'react-native';
import { shadow } from '../../helpers';

class ButtonLG extends Component {
  render() {
    const { onClick, children } = this.props;

    return (
      <TouchableHighlight
        onPress={onClick}
        style={styles.button}
        underlayColor={'#0A84D0'}
      >
        <Text style={styles.buttonText}>{children}</Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  button: {
    borderRadius: 3,
    padding: 20,
    width:'46%',
    marginVertical: 10,
    marginTop: 10,
    
    backgroundColor: '#1B95E0',
    ...shadow,
  },
  buttonText: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 22,
    
    marginLeft: "auto",
    marginRight: "auto",
  },
});

export default ButtonLG;
