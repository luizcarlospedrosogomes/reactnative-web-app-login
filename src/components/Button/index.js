import ButtonLG from './ButtonLG';

export default ButtonLG;
