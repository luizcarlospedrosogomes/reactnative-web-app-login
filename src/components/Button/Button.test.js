import { Platform, TouchableHighlight } from 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Button from './Button';

const onClick = jest.fn();
describe('Button', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<Button onClick={onClick}/>);
    expect(tree.toJSON()).toBeTruthy();
  });

  it('example snapshot test on - ' + Platform.OS, () => {
    const tree = renderer.create(<Button onClick={onClick}/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('onClick called when onPress fires', () => {
    onClick.mock.calls = [];
    const tree = renderer.create(<Button onClick={onClick}/>);
    tree.root.findByType(TouchableHighlight).props.onPress();
    expect(onClick.mock.calls).toEqual([[]]);
  });
});
