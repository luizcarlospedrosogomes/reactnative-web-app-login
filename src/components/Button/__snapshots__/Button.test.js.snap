// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button example snapshot test on - web 1`] = `
<div
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "MozUserSelect": "none",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 0,
      "WebkitUserSelect": "none",
      "alignItems": "stretch",
      "backgroundColor": "rgba(27,149,224,1.00)",
      "borderBottomLeftRadius": "3px",
      "borderBottomRightRadius": "3px",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopLeftRadius": "3px",
      "borderTopRightRadius": "3px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxShadow": "1px 1px 3px rgba(204,204,204,0.50)",
      "boxSizing": "border-box",
      "cursor": "pointer",
      "display": "flex",
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "marginBottom": "10px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "10px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 0,
      "msFlexPreferredSize": "auto",
      "msTouchAction": "manipulation",
      "msUserSelect": "none",
      "paddingBottom": "20px",
      "paddingLeft": "20px",
      "paddingRight": "20px",
      "paddingTop": "20px",
      "position": "relative",
      "touchAction": "manipulation",
      "userSelect": "none",
      "zIndex": 0,
    }
  }
  tabIndex="0"
>
  <div
    dir="auto"
    style={
      Object {
        "borderBottomWidth": "0px",
        "borderLeftWidth": "0px",
        "borderRightWidth": "0px",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "rgba(255,255,255,1.00)",
        "display": "inline",
        "fontFamily": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Ubuntu, \\"Helvetica Neue\\", sans-serif",
        "fontSize": "16px",
        "fontStyle": "inherit",
        "fontVariant": "inherit",
        "fontWeight": "bold",
        "lineHeight": "inherit",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "textDecoration": "none",
        "whiteSpace": "pre-wrap",
        "wordWrap": "break-word",
      }
    }
  />
</div>
`;
